tests/test_utils_http.py

```diff
@@ -311,3 +312,23 @@ def _is_uuid(string: str) -> bool:
 )
 def test_fix_hf_endpoint_in_url(base_url: str, endpoint: Optional[str], expected_url: str) -> None:
     assert fix_hf_endpoint_in_url(base_url, endpoint) == expected_url
+
+
+def test_adjust_range_header():
+    # Basic cases
+    assert _adjust_range_header(None, 10) == "bytes=10-"
+    assert _adjust_range_header("bytes=0-100", 10) == "bytes=10-100"
+    assert _adjust_range_header("bytes=-100", 10) == "bytes=-90"
```

Comment:
My understanding is that we should request starting from "10", not until "90"

```suggestion
    assert _adjust_range_header("bytes=-100", 10) == "bytes=10-100"
```

Reply:
Here, the original request bytes=-100 means "get the last 100 bytes", we've already downloaded 10 bytes, so now we only need the remaining 90 bytes from the end. If my understanding is correct, bytes=-90 correctly requests "the last 90 bytes".

Reply:
yes, exactly, `-100` is like `data[-100:]` in python, not `data[:100]`, so the expected result should be `bytes=-90`